srcJava/com/rti/perftest/ddsimpl/RTIDDSImpl.java

```diff
@@ -1634,67 +1645,78 @@ else if ("-noPositiveAcks".toLowerCase().startsWith(argv[i].toLowerCase())) {
         }
 
-        if (_IsAsynchronous && _batchSize > 0) {
-            System.err.println(
-                        "Batching cannot be used with asynchronous writing.");
-            return false;
+        /* Check if we need to enable Large Data. This works also for -scan */
+        if (_dataLen > Math.min(
+                MAX_SYNCHRONOUS_SIZE.VALUE,
+                MAX_BOUNDED_SEQ_SIZE.VALUE)) {
+            _isLargeData = true;
+            if (_useUnbounded == 0) {
+                _useUnbounded = MAX_BOUNDED_SEQ_SIZE.VALUE;
+            }
+        } else { /* No Large Data */
+            _useUnbounded = 0;
+            _isLargeData = false;
         }
 
-        if (_isScan) {
-            _dataLen = _scan_max_size;
-            // Check if large data or small data
-            if (_scan_max_size > Math.min(MAX_SYNCHRONOUS_SIZE.VALUE,MAX_BOUNDED_SEQ_SIZE.VALUE)) {
-                if (_useUnbounded == 0) {
-                    _useUnbounded = MAX_BOUNDED_SEQ_SIZE.VALUE;
+        /* If we are using batching */
+        if (_batchSize > 0) {
+
+            /* We will not use batching for a latency test */
+            if (_latencyTest) {
+                if (isBatchSizeProvided) {
+                    System.err.println(
+                            "Batching cannot be used in a Latency test.");
+                    return false;
+                }
+                else {
+                    _batchSize = 0; // Disable Batching
                 }
-                _isLargeData = true;
-            } else {
-                _useUnbounded = 0;
-                _isLargeData = false;
             }
+
+            /* Check if using asynchronous */
+            if (_IsAsynchronous) {
+                System.err.println(
+                        "Batching cannot be used with asynchronous writing.\n");
+                return false;
+            }
+
             /*
-             * If not Scan, compare sizes of Batching and dataLen At this point
-             * we have checked that we are not in a latency test mode, therefore
-             * we are sure we are in throughput test mode, where we do want to
-             * enable batching by default in certain cases
-             */
-        } else if (_batchSize > 0 && _batchSize < _dataLen * 2){
-            /*
-             * We don't want to use batching if the batch size is not large
-             * enough to contain at least two samples (in this case we avoid the
-             * checking at the middleware level).
+             * Large Data + batching cannot be set. But batching is enabled by default,
+             * so in that case, we just disabled batching, else, the customer set it up,
+             * so we explitly fail
              */
-            if (isBatchSizeProvided) {
+            if (_isLargeData) {
+                if (isBatchSizeProvided) {
+                    System.err.println(
+                        "Batching cannot be used with asynchronous writing.");
```

Comment:
add 4 spaces

Reply:
Done, actually it was "Batching cannot be used with Large Data." I fixed it.